packages/@aws-cdk/aws-scheduler-alpha/lib/target.ts

```diff
@@ -0,0 +1,277 @@
+ * @internal
+ */
+export function addToDeadLetterQueueResourcePolicy(schedule: ISchedule, queue: sqs.IQueue) {
+  if (!sameEnvDimension(schedule.env.region, queue.env.region)) {
+    throw new Error(`Cannot assign Dead Letter Queue in region ${queue.env.region} to the schedule ${Names.nodeUniqueId(schedule.node)} in region ${schedule.env.region}. Both the queue and the schedule must be in the same region.`);
+  }
+
+  // Skip Resource Policy creation if the Queue is not in the same account.
+  // There is no way to add a target onto an imported schedule, so we can assume we will run the following code only
+  // in the account where the schedule is created.
+  if (sameEnvDimension(schedule.env.account, queue.env.account)) {
+    const policyStatementId = `AllowSchedule${Names.nodeUniqueId(schedule.node)}`;
+
+    queue.addToResourcePolicy(new iam.PolicyStatement({
+      sid: policyStatementId,
+      principals: [new iam.ServicePrincipal('events.amazonaws.com')],
+      effect: iam.Effect.ALLOW,
+      actions: ['sqs:SendMessage'],
+      resources: [queue.queueArn],
+    }));
+  } else {
+    Annotations.of(schedule).addWarning(`Cannot add a resource policy to your dead letter queue associated with schedule ${schedule.scheduleName} because the queue is in a different account. You must add the resource policy manually to the dead letter queue in account ${queue.env.account}.`);
+  }
+}
+
+/**
+ * Whether two string probably contain the same environment dimension (region or account)
+ *
+ * Used to compare either accounts or regions, and also returns true if both
+ * are unresolved (in which case both are expted to be "current region" or "current account").
+ * @internal
+ */
+function sameEnvDimension(dim1: string, dim2: string) {
+  return [TokenComparison.SAME, TokenComparison.BOTH_UNRESOLVED].includes(Token.compareStrings(dim1, dim2));
+}
+
+export function renderRetryPolicy(maximumEventAge: Duration | undefined, maximumRetryAttempts: number | undefined): CfnSchedule.RetryPolicyProperty {
+  const maxMaxAge = Duration.days(1).toSeconds();
+  const minMaxAge = Duration.minutes(15).toSeconds();
+  let maxAge: number = maxMaxAge;
+  if (maximumEventAge) {
+    maxAge = maximumEventAge.toSeconds({ integral: true });
+    if (maxAge > maxMaxAge) {
+      throw new Error('Maximum event age is 1 day');
+    }
+    if (maxAge < minMaxAge) {
+      throw new Error('Minimum event age is 15 minutes');
+    }
+  };
+  let maxAttempts = 185;
+  if (typeof maximumRetryAttempts != 'undefined') {
+    if (maximumRetryAttempts < 0) {
+      throw Error('Number of retry attempts should be greater or equal than 0');
+    }
+    if (maximumRetryAttempts > 185) {
+      throw Error('Number of retry attempts should be less or equal than 185');
+    }
+    maxAttempts = maximumRetryAttempts;
+  }
+  return {
+    maximumEventAgeInSeconds: maxAge,
+    maximumRetryAttempts: maxAttempts,
+  };
+}
+
+export namespace targets {
```

Comment:
i thought we decided on putting these in `aws-scheduler-targets-alpha`?

Reply:
Yeah, you are right. I just wanted to get the constructs right for this commit to get feedback. Forgot to mention in the review notes. Will put the lambda targets in the separate package and the readme with the next commit.